packages/elements/src/components/Spreadsheet/utils.tsx

```diff
@@ -51,6 +64,40 @@ export const convertToCompatibleData = (parsedResult): Cell[][] =>
 /**
  * Convert back from Cell[][] to string[][]
  */
-export const convertDataToCsv = (data: Cell[][]): string[][] => data.map(rowArray => rowArray.map(({ value }) => value))
+export const convertDataToCsv = (data: Cell[][]): string[][] =>
+  data.map(rowArray => rowArray.map(({ value }) => value)) as string[][]
 
-export const hideContextMenu = (prev: ContextMenuProp): ContextMenuProp => ({ ...prev, visible: false })
+// Diffing algorithm to find differences between data array
+export const changedCellsGenerate = (newData?: Cell[][], oldData?: Cell[][]): ChangedCells => {
```

Comment:
Worth keeping an eye on how this performs - loops within loops can be really heavy. Might need to user a worker if it gets slow...

Reply:
I will keep an eye on it, maybe we can implement a better algorithm or using Web Worker as you said sounds cool.